packages/react-next/src/components/Slider/Slider.base.tsx

```diff
@@ -294,8 +296,11 @@ export class SliderBase extends React.Component<ISliderProps, ISliderState> impl
     }
     return currentPosition;
   }
+
   private _updateValue(value: number, renderedValue: number): void {
     const { step, snapToStep } = this.props;
+
+    // TODO this should be memoized based on step since it will only change when step changes
```

Comment:
I think that this is an interesting question. You're probably right that it's not too expensive but should it seems like it should only need to get set rarely. Do we want there to be a standard of memoizing/converting these things into state? This will happen automagically if when it get's converted to hooks if something useEffect is used.

Reply:
For now I'm going to leave this, and we can decide what we want later.

Reply:
Using state to cache expensive calculations has always been an anti-pattern, but people have done it a lot anyway due to the lack of standard/easy patterns around memoization. Doing memoization right will become a lot easier with function components with `React.useMemo`. But memoization itself does have a cost, so it's still not something that should be used in all cases.

(FYI since we're on the topic: it's good to keep in mind that [per the React docs](https://reactjs.org/docs/hooks-reference.html#usememo), `useMemo` is a performance optimization, *not* a semantic guarantee--see the docs for more on that, but basically, it means it doesn't guarantee that a calculation won't be re-run. Practically this would only cause problems for exceptionally expensive calculations or calculations with side effects, the latter of which should be avoided anyway.)